addons/sql_shell/models/sql_shell.py

```diff
@@ -0,0 +1,23 @@
+from odoo import models, fields
+
+
+class SQLShell(models.TransientModel):
+    _name = 'sql.shell'
+    _description = 'Shell'
+
+    query = fields.Text()
+    result = fields.Json(readonly=True)
+
+    def run(self):
+        try:
+            with self.env.cr.savepoint():
+                self.env.cr.execute(self.query)
+                self.result = {
+                    'header': [d.name for d in self.env.cr.description],
+                    'rows': self.env.cr.fetchall(),
```

Comment:
some request (update, delete) doesn't return rows

Reply:
note :  maybe add a limit to prevent browser crash
`'rows': self.env.cr.fetchall()[:1000],`

Reply:
> some request (update, delete) doesn't return rows

They return how many rows were updated, for instance

Reply:
> note :  maybe add a limit to prevent browser crash
> `'rows': self.env.cr.fetchall()[:1000],`

This depends on the browser/computer. Use at your own risks.
The purpose is not to fetch a bajillion rows. The way it is (temporarily) currently implemented would make the server's memory explode too btw.

Reply:
My bad, I was too quick in testing, I fixed both remarks in the latest commit